app/ts/background/storageVariables.ts

```diff
@@ -157,9 +157,10 @@ export async function updateTabState(tabId: number, updateFunc: (prevState: TabS
 const pendingAccessRequestsSemaphore = new Semaphore(1)
 export async function updatePendingAccessRequests(updateFunc: (prevState: PendingAccessRequestArray) => Promise<PendingAccessRequestArray>) {
 	return await pendingAccessRequestsSemaphore.execute(async () => {
-		const pendingAccessRequests = await updateFunc(PendingAccessRequestArray.parse(await browserStorageLocalSingleGetWithDefault('pendingInterceptorAccessRequests', [])))
+		const previous = PendingAccessRequestArray.parse(await browserStorageLocalSingleGetWithDefault('pendingInterceptorAccessRequests', []))
+		const pendingAccessRequests = await updateFunc(previous)
 		await browserStorageLocalSet('pendingInterceptorAccessRequests', PendingAccessRequestArray.serialize(pendingAccessRequests) as string)
```

Comment:
Could use an assertion to ensure serialize returns string

Reply:
Good catch! The type here is wrong, I need to refactor these by making browserStorageLocalSet to take an object that includes the proper type (PendingAccessRequestArray) with the key "pendingInterceptorAccessRequests" and then performing the serialization just before storing then. Similar refactor that I did to json rpc method return values